ui/src/app/shared/utils.ts

```diff
@@ -105,5 +105,16 @@ export const Utils = {
     // return a namespace, never return null/undefined, defaults to "default"
     getNamespace(namespace: string) {
         return this.managedNamespace || namespace || this.currentNamespace || 'default';
+    },
+
+    // renders a template literal with context variables
+    renderTemplate(template: string, context: any) {
+        context.toEpoch = (datetime: string) => {
+            return Math.floor(new Date(datetime).getTime() / 1000);
+        }
+        return template.replace(new RegExp("\\$\{([^\{]+)\}", "g"), function(_unused, expression){
+            const f = new Function("context", "with(context) { return " + expression + "}");
```

Comment:
Could this introduce any kind of XSS vulnerabilities? Do you have any prior-art you can share?

Reply:
I think so. Haven't tried XSS but if the user sets the link to something like `${alert(status.startedAt)}`, the alert will be shown.

I'm trying to see if there is a lightweight template engine that we can use. But didn't find any of them supporting our`${}` notation. I'll look into it. Thanks!

Reply:
I don't think we can risk new XSS vulnerabilities. Maybe we could have allow-list of functions?

Reply:
Totally agree. I did some investigation and end up using a template engine instead. Will explain more in a later comment. Thanks!